lib/ruby_indexer/lib/ruby_indexer/index.rb

```diff
@@ -255,7 +255,7 @@ def resolve_method(method_name, receiver_name)
     # that doesn't exist, then we return the same UnresolvedAlias
     sig { params(entry: Entry::UnresolvedAlias).returns(T.any(Entry::Alias, Entry::UnresolvedAlias)) }
     def resolve_alias(entry)
-      target = resolve(entry.target, entry.nesting)
+      target = resolve(entry.target, Array(entry.namespace&.name))
```

Comment:
While this might work, the idea behind the second argument is for it to be an array containing each part of the name.

I.e.:
```ruby
"Foo::Bar::Baz"
# Should be
["Foo", "Bar", "Baz"]
```

What do you think about adding a method named `nesting` in `UnresolvedAlias` that just splits the name?

```ruby
class UnresolvedAlias
  def nesting
    @namespace.name.split("::")
  end
end
```

Reply:
Sure, I can make that change but I think this part of the indexer is fundamentally broken, and needs a refactor anyway. The problem is that Ruby nesting is not as simple as looking at each part of the namespace, since implicit namespaces work differently:
```ruby
module Foo
  A = 1
end

module Foo::Bar
  puts A
  # => uninitialized constant Foo::Bar::A (NameError)
end
```

For more reference see: https://www.writesoftwarewell.com/nested-modules-in-ruby/

Thus, what we really need to do is to store the "nesting" as Ruby would have and resolve against that nesting stack.

Reply:
I think that's already what happens in main with the naming stack. Unless I'm missing something, I don't think it's broken.

We don't split namespaces that are defined as `module Foo::Bar`, so I believe the resolution is correct. Although we should add a test for that specific example.

The changes in this PR might require some extra refactors to make it fully work. For example, if we found an alias like this

```ruby
module First
  class Foo::Bar
    ALIAS_CONST = SOME_OTHER_THING
  end
end
```

We correctly store the nesting as `["First", "Foo::Bar"]`. When resolving `SOME_OTHER_THING`, we'd then search it in `First::Foo::Bar`, then `First` and finally on the top level.

And the features that use `resolve` do send the right nesting. For example, definition would invoke it like `@index.resolve("SOME_OTHER_THING", ["First", "Foo::Bar"])` - since we also don't split those names when locating target nodes for features.

However, with these changes we no longer have a reference to the nesting where things are found for unresolved symbols. We only have the entries with their fully qualified names in the stack, which as you pointed out is not enough to reconstruct the nesting where they were originally found.

To make the change in this PR and not break unresolved aliases, I think we need to change the entries to store their nesting instead of the fully qualified name, so that we can work back the nesting for aliases. Either that or maintain the original naming stack, but that seems more wasteful.

Reply:
Right, that makes sense. That also means that we are missing a lot of test cases that should have broken because of my changes in this PR, exercising specifically those kinds of nesting cases.